src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/AbstractBitbucketEndpoint.java

```diff
@@ -0,0 +1,139 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.cloudbees.jenkins.plugins.bitbucket.endpoints;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import com.cloudbees.plugins.credentials.domains.URIRequirementBuilder;
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.AbstractDescribableImpl;
+import hudson.security.ACL;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+
+/**
+ * Represents a {@link BitbucketCloudEndpoint} or a {@link BitbucketServerEndpoint}.
+ *
+ * @since 2.2.0
+ */
+public abstract class AbstractBitbucketEndpoint extends AbstractDescribableImpl<AbstractBitbucketEndpoint> {
+
+    /**
+     * {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     */
+    private final boolean manageHooks;
+
+    /**
+     * The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for auto-management of hooks.
+     */
+    @CheckForNull
+    private final String credentialsId;
+
+    /**
+     * Constructor.
+     *
+     * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
+     * @param credentialsId The {@link StandardUsernamePasswordCredentials#getId()} of the credentials to use for
+     *                      auto-management of hooks.
+     */
+    AbstractBitbucketEndpoint(boolean manageHooks, @CheckForNull String credentialsId) {
+        this.manageHooks = manageHooks && StringUtils.isNotBlank(credentialsId);
+        this.credentialsId = manageHooks ? credentialsId : null;
+    }
+
+    /**
+     * Optional name to use to describe the end-point.
+     *
+     * @return the name to use for the end-point
+     */
+    @CheckForNull
+    public abstract String getDisplayName();
```

Comment:
Not a breaking change, because we can return for you the computed name if the name is not present or blank.

GitHub and BitBucket URLs are not going to be crazy long, because those services demand a root URL so it is basically just the hostname.

Users are going to want to see the URL that the display name corresponds to anyway...

> Which GitHub is `https://github3.example.com`? is that `Production` or `QA`?

So we need to add the URL to the display name to help the user

> Which GitHub is `https://github3.example.com`? is that `Production (https://github3.example.com)` or `QA (https://github2.example.com)`

This is because the user is used to the GIT URL.

They will know the host name